engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -181,16 +179,39 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
     val actionThreadPoolSize = databaseConfig.getIntOr("actionThreadPoolSize", dbNumThreads) min dbMaximumPoolSize
     Executors.newFixedThreadPool(actionThreadPoolSize)
   }
+
   private val actionExecutionContext: ExecutionContext = ExecutionContext.fromExecutor(
-    actionThreadPool, database.executor.executionContext.reportFailure)
+    actionThreadPool, database.executor.executionContext.reportFailure
+  )
+
+  private lazy val useSlickUpserts = dataAccess.driver.capabilities.contains(JdbcProfile.capabilities.insertOrUpdate)
 
   override def close(): Unit = {
     actionThreadPool.shutdown()
     database.close()
   }
 
-  // Run action with an outer transaction
-  private def runTransaction[R](action: DBIOAction[R, _ <: NoStream, _ <: Effect]): Future[R] = {
+  private def runTransactionWithRollbackRetry[R](action: DBIO[R], actorSystem: ActorSystem): Future[R] = {
```

Comment:
Ok, it just seemed a bit odd to inject akka stuff into slickDataAccess only to use a scheduler, but since we're probably going to actorify a lot of DB IO it's going to the right direction